File: frontend/src/types/businessRule.ts
export interface RuleCondition {
  field: string
  operator: string
  value: string
  logicalOp: "AND" | "OR" | null
}

export interface BusinessRule {
  id: string
  name: string
  conditions: RuleCondition[]
  action: string
  priority: number
  status: "Active" | "Inactive" | "Draft"
  executionCount: number
  country: string
  segment: string
  createdAt: string
}

File: backend/app/schemas/business_rule.py
from typing import Literal
from pydantic import BaseModel


class RuleCondition(BaseModel):
    field: str
    operator: str
    value: str
    logicalOp: Literal["AND", "OR", None] = None


class BusinessRule(BaseModel):
    id: str
    name: str
    conditions: list[RuleCondition]
    action: str
    priority: int
    status: Literal["Active", "Inactive", "Draft"]
    executionCount: int
    country: str
    segment: str
    createdAt: str
